src/Kestrel.Transport.Libuv/Internal/Listener.cs

```diff
@@ -181,7 +181,10 @@ private void OnConnection(UvStreamHandle listenSocket, int status)
 
         protected virtual void DispatchConnection(UvStreamHandle socket)
         {
-            var connection = new LibuvConnection(this, socket);
+            var connection = new LibuvConnection(socket, Log, Thread);
+
+            TransportContext.ConnectionDispatcher.OnConnection(connection);
```

Comment:
How is this going to flow once OnConnection starts returning a lifetime Task?

Reply:
I guess the transport doesn't need a task for the connection lifetime. I guess just rename to AcceptConnection or something like that.

Also, make sure we're confident that On/AcceptConnection doesn't throw since you moved all the calls out of a try catch.